easier68k/core/opcodes/Bcc.py

```diff
@@ -0,0 +1,358 @@
+        simulator.increment_program_counter(self.offset)
+        
+    def __str__(self):
+        """
+        converts command to a string
+        :return: string rep of command
+        """
+
+        return "{s} ${d}".format(
+                s = COND_CODE_TO_OPCODE.get(self.cond_code),
+                d = self.operand or (self.address + 2 + (self.size if self.size > 1 else 0) + self.offset))
+
+    @classmethod
+    def command_matches(cls, command: str) -> bool:
+        """
+        Checks whether a command string is an instance of this command type
+        :param command: The command string to check (e.g. 'MOVE.B', 'LEA', etc.)
+        :return: Whether the string is an instance of this command type
+        """
+        raise Exception("Branch Operation did not override parent class command_matches stub!")
+        #return
+
+    @classmethod
+    def get_word_length(cls, command: str, parameters: str) -> int:
+        """
+        Gets what the end length of this command will be in memory
+        :param command: The text of the command itself (e.g. "LEA", "MOVE.B", etc.)
+        :param parameters: The parameters after the command
+        :return: The length of the bytes in memory in words
+        """
+        
+        # opword
+        length = 2
+        
+        # BIG NOTE: I am explicitly ignoring size codes because there are no defined flippy boys
+        # in the opword for Bcc ops. Yeezy68k might let you define a size code, but even it
+        # will more or less ignore whatever you put (reduces unnessesary .l to .w)
+        
+        parameter = parameters.split(',')[0]
+        
+        # Check range of operand
+        parameter = parse_assembly_parameter(parameter.strip())
+        
+        if parameter.mode == EAMode.AWA:
+            length += 2
+        if parameter.mode == EAMode.ALA:
+            length += 4
+        
+        return length
+    
+    @classmethod
+    def is_valid(cls, command: str, parameters: str) -> (bool, list):
+        """
+        Tests whether the given command is valid
+
+        :param command: The command itself (e.g. 'MOVE.B', 'LEA', etc.)
+        :param parameters: The parameters after the command (such as the source and destination of a move)
+        :return: Whether the given command is valid and a list of issues/warnings encountered
+        """
+        
+        passes = True
+        errors = []
+        parameters = parameters.split(',')
+        
+        # See the BIG HORKIN' NOTE from get_word_length()
+        # check param length
+        if len(parameters) >2:
+            passes = False
+            errors.append("Too many operands for Branch Operation (TODO: list them here)")
+        
+        # check offset dest validity (odd read)
+        operand = parse_assembly_parameter(parameters[0].strip())
+        if operand.data % 2 == 1:
+            passes = False
+            errors.append("Reading word at {d} will cause crash".format(d=operand.data))
```

Comment:
if only we could target >3.6 we could use f strings here

Reply:
On one hand most distributions of linux ship 3.6
On the other, current stable is 3.7.2 iirc

Reply:
True, but do we want to break 3.5? It's not crucial, so maybe not.